packages/vercel-ai-provider/src/edgedb-provider.ts

```diff
@@ -0,0 +1,90 @@
+import { type Client, createClient } from "edgedb";
+import type { ResolvedConnectConfig } from "edgedb/dist/conUtils.js";
+import { getHTTPSCRAMAuth } from "edgedb/dist/httpScram.js";
+import { cryptoUtils } from "edgedb/dist/browserCrypto.js";
+import { getAuthenticatedFetch } from "edgedb/dist/utils.js";
+import type {
+  EmbeddingModelV1,
+  LanguageModelV1,
+  ProviderV1,
+} from "@ai-sdk/provider";
+import {
+  EdgeDBChatLanguageModel,
+  type EdgeDBLanguageModel,
+} from "./edgedb-chat-language-model";
+import type {
+  EdgeDBChatModelId,
+  EdgeDBChatSettings,
+} from "./edgedb-chat-settings";
+import { EdgeDBEmbeddingModel } from "./edgedb-embedding-model";
+import type {
+  EdgeDBEmbeddingModelId,
+  EdgeDBEmbeddingSettings,
+} from "./edgedb-embedding-settings";
+
+const httpSCRAMAuth = getHTTPSCRAMAuth(cryptoUtils);
+
+export interface EdgeDBProvider extends ProviderV1 {
+  (modelId: EdgeDBChatModelId | EdgeDBEmbeddingModelId): LanguageModelV1;
+
+  languageModel(
+    modelId: EdgeDBChatModelId,
+    settings?: EdgeDBChatSettings,
+  ): EdgeDBLanguageModel;
+
+  textEmbeddingModel: (
+    modelId: EdgeDBEmbeddingModelId,
+    settings?: EdgeDBEmbeddingSettings,
+  ) => EmbeddingModelV1<string>;
+}
+
+export async function createEdgeDB(client: Client): Promise<EdgeDBProvider> {
+  const connectConfig: ResolvedConnectConfig = (
+    await (client as any).pool._getNormalizedConnectConfig()
+  ).connectionParams;
```

Comment:
I think we have a new API for getting these on the Client object. Like `getConnectionParams` or something?

Reply:
I updated it.